tests/gadget-zero/test_gadget0.py

```diff
@@ -369,3 +408,75 @@ def test_aligned(self):
     def test_unaligned(self):
         self.set_unaligned()
         self.do_readwrite()
+
+
+class TestControlTransfer_Delays(unittest.TestCase):
+    """
+    Test cases for
+    https://github.com/libopencm3/libopencm3/issue/668
+    """
+
+    def setUp(self):
+        self.dev = usb.core.find(idVendor=VENDOR_ID, idProduct=PRODUCT_ID, custom_match=find_by_serial(DUT_SERIAL))
+        self.assertIsNotNone(self.dev, "Couldn't find locm3 gadget0 device")
+
+        self.cfg = uu.find_descriptor(self.dev, bConfigurationValue=4)
+        self.assertIsNotNone(self.cfg, "Config 4 should exist")
+        self.dev.set_configuration(self.cfg)
+        self.req     = uu.CTRL_IN  | uu.CTRL_TYPE_VENDOR | uu.CTRL_RECIPIENT_INTERFACE
+        self.req_out = uu.CTRL_OUT | uu.CTRL_TYPE_VENDOR | uu.CTRL_RECIPIENT_INTERFACE
+        self.intr_out = 1
+        # send dummy packages on interrupt port to get in sync
+        # (alternating bit protocol)
+        # FIXME: Why is this necessary?
```

Comment:
Something on the hardware level.  The alternating bit protocol is not visible by the software but implemented in the STMF2 in hardware.  Due to the timeout one of the parties doesn't toggle the bit (I don't remember which), so the next message is just ignored.